src/sqlshim.php

```diff
@@ -248,13 +248,38 @@ private static function calc_prec_scale( $prec, $scale )
   private static function typify( $row )
   {
     $i = 0;
-    foreach ( $row as $col=>&$val )
-    {
-      $val = self::guesstype($val);
-      $i++;
+    foreach ($row as &$value) {
+        //DBLIB database driver returns everything as strings, so this converts num's back to the correct data type
+        $value = self::convertDataType($value);
     }
     return $row;
   }
+  
+  private static function convertDataType($string){
```

Comment:
it probably wasn't working due to this line: https://github.com/radsectors/sqlshim/blob/de3e945fc54e58f8294f12515d9247e575e5605a/src/sqlshim.php#L280

You'd just have to set the option 'autotype_fields' to true in your sqlshim::init() array to get it to work. I have it off by default until I can come up with a better option.
```php
sqlshim::init(['autotype_fields']);
```

Speaking of better options... I found this: http://php.net/manual/en/ref.pdo-dblib.php#89827
may be on to something.